src/interfaces/records-read.ts

```diff
@@ -57,6 +67,32 @@ export class RecordsRead extends Message<RecordsReadMessage> {
     return new RecordsRead(message);
   }
 
+  /**
+   * Creates a Records interface Filter using either the `recordId` or `protocol` & `protocolPath` of the incoming Message.
+   * Also filters on an optional `parentId` for `protocolPath` based reads
+   * @param descriptor message descriptor with optional properties `recordId`, `protocol` and `protocolPath`
+   *
+   * @returns {Filter} with a Records interface as well as the appropriate filter params
+   * @throws {DwnError} if `recordId` or `protocol` and `protocolPath` are missing
+   */
+  public static createFilter(descriptor: Partial<Omit<RecordsReadDescriptor, 'method'>>): Filter {
```

Comment:
The ugly types like `Partial<Omit<RecordsReadDescriptor, 'method'>>` would typically suggest improvement or refactoring somewhere.

Perhaps this can be improved after the merge of #474

Reply:
Yeah, I really didn't love that either.

And yup! Looking at #474, it would eliminate this from being used outside the context of `RecordsRead`.

Reply:
Sounds good, let's resolve the conflicts now that #474 is in, I'll do a final pass on the PR!